donjuan/randomizer.py

```diff
@@ -41,3 +44,30 @@ def randomize_grid(self, grid: Grid) -> None:
             for j in range(grid.n_cols):
                 self.randomize_cell(grid.cells[i][j])
         return
+class DoorSpaceRandomizer(Randomizer):
+    """
+    Randomly set the :attrs: of the DoorSpace and select a DoorSpace type (Door, Archway, Portcullis).
+    """
+    def randomize_doorspace(self, door_space:DoorSpace):
+        #
+        door_types = [Door, Portcullis, Archway]
+        material_types = ["wood","metal","stone"]
+        #randomize all possible attributes
+        blocked = bool(random.randint(0,1))
+        secret = bool(random.randint(0,1))
+        locked = bool(random.randint(0,1))
+        broken = bool(random.randint(0,1))
+        jammed = bool(random.randint(0,1))
+        closed = bool(random.randint(0,1))
+        material = material_types[random.randint(0,len(material_types) - 1)]
+        #pick a type and make a new instance of it
+        selected_type = door_types[random.randint(0,len(door_types) - 1)]()
+
+        if(isinstance(selected_type, Door)):
+            door_space = Door(secret=secret,locked=locked,closed=closed,jammed=jammed,blocked=blocked,broken=broken,material=material)
```

Comment:
We don't want to replace the object with another (it wouldn't work anyway, because pass by reference). We just want to set the attributes.

```suggestion
            door_space.secret=secret
            door_space.locked=locked
            door_space.closed=closed
            door_space.jammed=jammed
            door_space.blocked=blocked
            door_space.broken=broken
            door_space.material=material
```

same goes for the portcullis and archway cases.